controllers/machine_controller.go

```diff
@@ -238,6 +249,14 @@ func (r *MachineReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cl
 	logger := r.Log.WithValues("machine", m.Name, "namespace", m.Namespace)
 	logger = logger.WithValues("cluster", cluster.Name)
 
+	if err := r.watchClusterNodes(ctx, cluster); err != nil {
+		// Remote cluster may not be initialized yet.
+		if !strings.Contains(err.Error(), remote.ErrCreatingRemoteClusterCache.Error()) {
+			logger.Error(err, "error watching nodes on target cluster")
+			return ctrl.Result{}, err
+		}
```

Comment:
Is it right that we are ignoring all the other types if errors?

Reply:
Also, in https://github.com/kubernetes-sigs/cluster-api/blob/f881755d4897695b28b4759ec4253c423ba67e56/controllers/machinehealthcheck_controller.go#L199-L202 there is a different error management
Is this expected? If yes should we document why?

Reply:
Connection to remote cluster fails due to remote cluster control plane is not up yet. We need to continue reconcile in that scenario. There is one more error return from `Tracker.Watch` which is `error creating watch`, but I thought if the control plane is up and able to connect it, we should return error if that happens.

I am thinking out loud, another option is to check if infrastructure ready condition then start watching nodes?

Reply:
In MHC, it makes sense to wait until connection to remote cluster succeeds to continue reconcile.
But in Machine controller we should continue even if remote cluster is not up yet, e.g., when KCP creates the first Machine.

Reply:
Thanks for the explanation
I think it makes sense to create the watch only if `cluster.Status.ControlPlaneInitialized == true` (we need the API server in the workload cluster to be up & running, not only the infrastructure being ready)